File: 226-invert-binary-tree/226-invert-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:        

        
        if root is None: return root
        
        def recur(node):
            if node is None: return None
            newTree = TreeNode(node.val)
            newTree.left = recur(node.right)
            newTree.right = recur(node.left)
            
            return newTree
        
        
        
        return recur(root)
        

File: 226-invert-binary-tree/226-invert-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {TreeNode}
 */
var invertTree = function(root) {
    if(root===null){
        return root
    }
    
    function helper(node){
        if(node===null) return null
        
        let newTree = new TreeNode(node.val)      
        newTree.left = helper(node.right)
        newTree.right = helper(node.left)
        
        
        return newTree
    }
    
    return helper(root)
};